### PR TITLE
feat/cart and wishlist

### PR DESCRIPTION
**Pull request by adith-a-danthi**
## Overview

- Add Cart Context + Reducer
- Add Wishlist Context + Reducer
- New Components
  - Cart Item
  - Wishlist Item
  - Cart Summary
- New Pages / Views
  - Cart
  - Wishlist

### CODE REVIEW

**Review by Mohitkumar33 — COMMENTED** [review]
So you are doing this without api call

**Review by omjadhav85 — APPROVED** [review]
Clean and well structured code. Well placed comments wherever required. Good to merge!

**Comment by adith-a-danthi** [comment]
> So you are doing this without api call

@Mohitkumar33 Haven't integrated the auth APIs yet, so will do the API integration for cart and wishlist after that.

File: src/components/CartItem/CartItem.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { useCart } from '../../contexts/cart-context';
+import { useWishlist } from '../../contexts/wishlist-context';
```
Comment (faisal-kursheedali) [inline src/components/CartItem/CartItem.jsx:2]: Looks good to me, good to go.

**Review by Mohitkumar33 — APPROVED** [review]

File: src/components/CartItem/CartItem.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    dispatchWishlist({ type: 'ADD_TO_WISHLIST', payload: product });
+  };
+
+  return (
+    <div className="card outlined-card cart-item flex">
+      <img className="card-img" src={imageURL} alt={title} />
+
+      <div className="card-content flex-1">
+        <div className="heading-4">{title}</div>
+
```
Comment (Bharatrathod-81) [inline src/components/CartItem/CartItem.jsx:23]: perfectly divided the things by comment that awesome

**Review by Bharatrathod-81 — APPROVED** [review]

**Pull request merged**